### DISCUSSION TITLE
不能运行data("NHANES_20072008")，提示Warning message: In data("NHANES_20072008") : 没有‘NHANES_20072008’这个数据集

### DISCUSSION THREAD

**Discussion by JustGNest** [jamesjiadazhan/dietaryindex#231] [Q&A] [answered]
已经下载NHANES_20072008.rda数据，并且已经load，右侧environment下也可以看到Data NHANES_20072008 Large list(9 elements)。但就是没办法运行data("NHANES_20072008")。
然后放弃作者提供的数据包，自行下载数据，按
library(dietaryindex)
library(haven)
DEMO_PATH = read_xpt(file ='DEMO_E.XPT')#人口
FPED_PATH_1 = read_sas('fped_dr1tot_0708.sas7bdat')#第一天数据
NUTRIENT_PATH_1 = read_xpt(file ='DR1TOT_E.XPT')#第一天数据
FPED_PATH_2 = read_sas('fped_dr2tot_0708.sas7bdat')#第二天数据
NUTRIENT_PATH_2 = read_xpt(file ='DR2TOT_E.XPT')#第二天数据
再运行DASH_NHANES_FPED(FPED_PATH=FPED_PATH_1, NUTRIENT_PATH=NUTRIENT_PATH_1, FPED_PATH2=FPED_PATH_2, NUTRIENT_PATH2=NUTRIENT_PATH_2）
也报错，但是运行
DASHI_NHANES_FPED(NUTRIENT_PATH=NUTRIENT_PATH_1, NUTRIENT_PATH2=NUTRIENT_PATH_2)
倒是成功的。求大佬帮助
![image](https://github.com/jamesjiadazhan/dietaryindex/assets/35002858/fdd16bd1-eb17-49a1-b206-e4cd43b68222)

**Comment by jamesjiadazhan** [comment, marked as answer]
data("NHANES_20072008") only works if the data is available within the dietaryindex package. If you download and load the data, there is no need to use the "data" function.

You're using the wrong NHANES datasets for the DASH function, which requires individual NHANES files. I updated the readme page as below: (https://github.com/jamesjiadazhan/dietaryindex). Please check the difference between NHANES files and what the DASH function requires.

Let me know if you have other questions. If possible, please give our work a Star on GitHub and cite our work when you publish the work. It helps recognize and appreciate our intellectual efforts and contributions.
- Jiada James Zhan, Rebecca A Hodge, Anne Dunlop, et al. Dietaryindex: A User-Friendly and Versatile R Package for Standardizing Dietary Pattern Analysis in Epidemiological and Clinical Studies. bioRxiv. Published online August 07, 2023:2023.08.07.548466. doi:10.1101/2023.08.07.548466
